src/lib.rs

```diff
@@ -2,9 +2,20 @@
 // This is an HTML parser. HTML can be untrusted input from the internet.
 #![forbid(unsafe_code)]
 #![doc = include_str!("../README.md")]
+#![warn(clippy::all, clippy::pedantic)]
+#![warn(
+    absolute_paths_not_starting_with_crate,
+    rustdoc::invalid_html_tags,
+    missing_copy_implementations,
+    missing_debug_implementations,
+    semicolon_in_expressions_from_macros,
+    unreachable_pub,
+    unused_extern_crates,
+    variant_size_differences,
+    clippy::missing_const_for_fn
```

Comment:
I think this lint is bad, just because my function can be const doesn't mean I want to provide that api guarantee

Reply:
Making eligible functions const allows us or downstream library users use them in const context.

And if you wish you can opt-out this link on a per-function or a per-module base.

Reply:
I'm referring to this concern: https://github.com/rust-lang/rust-clippy/issues/7121

Just because a (public) function can be `const` doesn't mean I want to make this API guarantee (because removing `const` again is a breaking change) -- it's especially a commitment to expose a `const fn` when so few things are `const` still, because it limits the implementation

I don't have a good example in this crate as it seems you had to fix a single internal function, but I think the clippy lint is kind of misguided